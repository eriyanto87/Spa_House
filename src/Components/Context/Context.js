import React from "react";

const Context = React.createContext({
  // users: [],
  // cart: [],
  // treatments: [],
  // addUser: () => {},
  // addCart: () => {},
  //   getAllTreatments: () => {},
});

export default Context;
