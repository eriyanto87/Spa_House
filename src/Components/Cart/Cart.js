import React from "react";
import Context from "../Context/Context";
import Nav from "../Nav/Nav";
import Footer from "../Footer/Footer";

export default function Cart(props) {
  const goBack = () => props.history.goBack();

  // const onSubmit (get comments)
  //go to confirmation page
  // const comment
  //POST
  const onSubmit = (e, context) => {
    e.preventDefault();
    console.log("cart button works");

    const comments = e.target.comments.value;
    console.log(comments);
    console.log(context.users.length - 1, "latest user");
    const latestUserIndex = context.users.length - 1;

    const cartData = {
      user_id: context.users[latestUserIndex].id,
      treatment_id: context.treatment,
      appointment_date: context.databaseDate,
    };

    console.log(cartData);
    context.addConfirmation(cartData);
    props.history.push("/confirmation");
  };

  return (
    <Context.Consumer>
      {(context) => {
        console.log(context);

        const name = context.treatments.filter(
          (t) => t.id == context.treatment
        );
        return (
          <div>
            <header>
              <Nav />
            </header>

            <section>
              <h3>Cart</h3>
              <form onSubmit={(e) => onSubmit(e, context)}>
                <p className="clientInfo">{name[0].display_name}</p>
                <h4> Your appointment date and time:</h4>
                <p className="clientInfo">
                  {context.displayDate} {context.displayTime}
                </p>
                <div>
                  <p>
                    <label>Comments:</label>
                  </p>
                  <textarea
                    name="comments"
                    placeholder="Is there anything I need to know prior to the service? etc: any injuries? are you pregnant? any skin issues such as skin burn, etc"
                  />
                  <h4>Total:</h4>
                  <h3 className="clientInfo">${name[0].price}</h3>
                </div>
                <hr />
                <button onClick={goBack}>Back</button>{" "}
                <button type="submit">Confirmation</button>
              </form>
            </section>
            <footer>
              <Footer />
            </footer>
          </div>
        );
      }}
    </Context.Consumer>
  );
}
