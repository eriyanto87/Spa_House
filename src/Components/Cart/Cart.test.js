import React from "react";
import ReactDOM from "react-dom";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import Context from "../Context/Context";
import Cart from "./Cart";

describe("Cart component", () => {
  const contextValues = {
    users: [
      {
        user_name: "",
      },
    ],
    addConfirmation: () => {},
    treatments: [
      {
        display_name: "",
      },
    ],
    displayTime: "",
    displayDate: "",
  };

  it("Cart Snapshot", () => {
    const tree = renderer
      .create(
        <BrowserRouter>
          <Context.Provider value={contextValues}>
            <Cart />
          </Context.Provider>
        </BrowserRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <BrowserRouter>
        <Context.Provider value={contextValues}>
          <Cart />
        </Context.Provider>
      </BrowserRouter>,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });
});
