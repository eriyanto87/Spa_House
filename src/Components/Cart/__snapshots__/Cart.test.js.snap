// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart component Cart Snapshot 1`] = `
<div>
  <header>
    <div
      className="nav"
    >
      <ul>
        <li>
          <a
            href="/"
            onClick={[Function]}
          >
            Home 
          </a>
        </li>
        <li>
          <a
            href="/about"
            onClick={[Function]}
          >
            About 
          </a>
        </li>
        <li>
          <a
            href="/treatments"
            onClick={[Function]}
          >
            Treatments 
          </a>
        </li>
        <li>
          <a
            href="/pricing"
            onClick={[Function]}
          >
            Pricing 
          </a>
        </li>
        <li>
          <a
            href="/contact"
            onClick={[Function]}
          >
            Contact Us 
          </a>
        </li>
        <li>
          <a
            href="/gallery"
            onClick={[Function]}
          >
            Gallery
          </a>
        </li>
      </ul>
    </div>
  </header>
  <section>
    <h3>
      Cart
    </h3>
    <form
      onSubmit={[Function]}
    >
      <p
        className="clientInfo"
      >
        
      </p>
      <h4>
         Your appointment date and time:
      </h4>
      <p
        className="clientInfo"
      >
        
         
        
      </p>
      <div>
        <p>
          <label>
            Comments:
          </label>
        </p>
        <textarea
          name="comments"
          placeholder="Is there anything I need to know prior to the service? etc: any injuries? are you pregnant? any skin issues such as skin burn, etc"
        />
        <h4>
          Total:
        </h4>
        <h3
          className="clientInfo"
        >
          $
        </h3>
      </div>
      <hr />
      <button
        onClick={[Function]}
      >
        Back
      </button>
       
      <button
        type="submit"
      >
        Confirmation
      </button>
    </form>
  </section>
  <footer>
    <div>
      <p>
        Â© 2020
      </p>
    </div>
  </footer>
</div>
`;
